### PR TITLE
drivers: implement sec driver

### PR DESCRIPTION
**Pull request by xiaoxuZeng**
`Sec` is a Security Engine for cipher, digest, HMAC, authenc.

`Qm` is a queue management device include by Sec.

This `PR` just implement sec cipher function. 

digest, HMAC, authenc. will add on next `PR`, when we finished development and test.

### CODE REVIEW

File: core/drivers/crypto/kunpeng/qm/qm.h [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	 * cqe_size : 4
+	 * rsv(stash_nid/stash_en) : 16
+	 */
+	uint32_t dw3;
+	uint16_t rand_data;
+	uint16_t rsv0;
+	uint16_t pasid;
+	/*
+	 * pasid_en : 1
+	 * rsv 锛�
```
Comment (xiaoxuZeng) [inline core/drivers/crypto/kunpeng/qm/qm.h:102]: There is a garbled character, I will fix now.

**Comment by jenswi-linaro** [comment]
A few early comments:
I'd prefer a common prefix for all symbols below `core/drivers/crypto/kunpeng`, compare with what the other crypto drivers have done. You don't necessarily have to use `kunpeng_` as a prefix, some abbreviation may also suffice.

Please fix the checkpatch issues, if there are some you can't fix nicely bring them up for discussion and we'll see what we can do about it.

Bitfields in structs as a way of interacting with hardware is not portable and should be avoided.

**Review by etienne-lms — COMMENTED** [review]
Coding style and minor comment taht can be addressed straight.

File: core/drivers/crypto/kunpeng/sec/sec_cipher.c [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+	case TEE_ALG_DES3_CBC_NOPAD:
+	case TEE_ALG_SM4_CBC_NOPAD:
+	case TEE_ALG_SM4_ECB_NOPAD:
+	case TEE_ALG_SM4_CTR:
+		break;
+	default:
+		return TEE_ERROR_NOT_IMPLEMENTED;
+	}
+
+	return 0;
```
[the remark this answers is not in the document]
Comment (xiaoxuZeng) [reply core/drivers/crypto/kunpeng/sec/sec_cipher.c:446]: Good review. I will check all of it. thanks

**Pull request closed without merging**